dropshot/tests/fail/bad_endpoint2.stderr

```diff
@@ -18,23 +18,3 @@ error: Expected a non-receiver argument
    |
 13 | async fn bad_endpoint(self) -> Result<HttpResponseOk<()>, HttpError> {
    |                       ^^^^
-
-error: `self` parameter is only allowed in associated functions
-  --> tests/fail/bad_endpoint2.rs:13:23
-   |
-13 | async fn bad_endpoint(self) -> Result<HttpResponseOk<()>, HttpError> {
-   |                       ^^^^ not semantically valid as function parameter
-   |
-   = note: associated functions are those in `impl` or `trait` definitions
-
-error[E0401]: can't use generic parameters from outer item
-  --> tests/fail/bad_endpoint2.rs:13:23
-   |
-9  | #[endpoint {
-   | ---------- `Self` type implicitly declared here, by this `impl`
-...
-13 | async fn bad_endpoint(self) -> Result<HttpResponseOk<()>, HttpError> {
-   |                       ^^^^
-   |                       |
-   |                       use of generic parameter from outer item
-   |                       refer to the type directly here instead
```

Comment:
I think these errors in particular are unhelpful, and detecting them syntactically is much cleaner.

Reply:
What  does "detecting them syntactically" mean and how does that relate to this change? Agreed that these errors are not helpful.

Reply:
By "syntactically" I just mean looking at the token stream and rejecting code within the proc macro, rather than "semantically", generating code that will fail to compile and relying on rustc's errors. (Is there a better term for this? I've always used syntactically/semantically for this in the context of proc macros)